lib/iris/plot.py

```diff
@@ -529,7 +533,11 @@ def _ensure_cartopy_axes_and_determine_kwargs(x_coord, y_coord, kwargs):
         cartopy_proj = ccrs.PlateCarree()
 
     # Ensure the current axes are a cartopy.mpl.GeoAxes instance.
-    _replace_axes_with_cartopy_axes(cartopy_proj)
+    if 'axes' not in kwargs:
+        _replace_axes_with_cartopy_axes(cartopy_proj)
+    elif not not isinstance(kwargs['axes'], cartopy.mpl.geoaxes.GeoAxes):
```

Comment:
double not. I'm not not sure what happens here...

This highlights the need for a test of this error I think.